drivers/ad5683/ad5683.h

```diff
@@ -0,0 +1,196 @@
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#ifndef _AD5683_H_
+#define _AD5683_H_
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <stdint.h>
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+#define FAILURE                    -1
+#define AD5683_CMD_NOP             0  // No operation.
+#define AD5683_CMD_WR_IN_REG       1  // Write Input Register.
+#define AD5683_CMD_UPDATE_DAC      2  // Update DAC Register.
+#define AD5683_CMD_WR_DAC_REG      3  // Write DAC and Input Register.
+#define AD5683_CMD_WR_CTRL_REG     4  // Write Control Register.
+#define AD5683_CMD_RB_IN_REG       5  // Readback input register.
+
+#define MAX_RESOLUTION             16 // Maximum resolution of the supported devices
+
+/* Power-down operation modes masks */
+#define AD5683_PWRM_NORMAL         0	// Power Mode Normal
+#define AD5683_PWRM_1K             1	// Power-down mode 1kOhm to GND
+#define AD5683_PWRM_100K           2	// Power-down mode 100kOhm to GND
+#define AD5683_PWRM_THREESTATE     3	// Three-State
+
+#define AD5683_DC_EN               1
```

Comment:
as a practice : for `<x>_EN` and `<x>_DIS` value pairs, you could just define the behavior of `0 = disable ; non-zero = enable`